### ISSUE TITLE
Add README and LICENSE to staging

### ISSUE THREAD

**Issue by nikhita** [kubernetes/kubernetes#54398]
From https://github.com/kubernetes/kubernetes/pull/54293#issuecomment-338594244 and https://github.com/kubernetes/sample-controller/pull/1#issuecomment-338594424, we should use the staging files instead of having some files authoritative in the external repo. Otherwise, we complicate the publishing process as it has to know which files come from the latter.

`README.md` and `LICENSE` are authoritative in external repos. We should move them to staging.

This is a placeholder issue to reference the removal of README and LICENSE from all external repos (and addition into staging).

/cc @sttts

**Comment by sttts** [comment]
It also creates confusion that some files are authorative in the published repos. Moreover, we cannot switch to an absolute publishing algorithm (easily) that doesn't work with cherry-picks. The former is considerably simpler than what is implemented in 700 lines of bash in https://github.com/kubernetes/test-infra/pull/4134.

**Comment by nikhita** [comment]
Addition of files to staging repos: https://github.com/kubernetes/kubernetes/pull/54399.

Removal of files from published repos:

1. api - https://github.com/kubernetes/api/pull/9
2. apiextensions-apiserver - https://github.com/kubernetes/apiextensions-apiserver/pull/18
3. apimachinery - https://github.com/kubernetes/apimachinery/pull/28
4. apiserver - https://github.com/kubernetes/apiserver/pull/22
5. code-generator - https://github.com/kubernetes/code-generator/pull/12
6. kube-aggregator - https://github.com/kubernetes/kube-aggregator/pull/10
7. metrics - https://github.com/kubernetes/metrics/pull/10
8. sample-apiserver- https://github.com/kubernetes/sample-apiserver/pull/16
9. sample-controller - https://github.com/kubernetes/sample-controller/pull/1

**Comment by lavalamp** [comment]
It makes sense to me that things should be 100% authoritative in staging, not 98%.

What about client-go?

**Comment by sttts** [comment]
> What about client-go?

Waiting for @caesarxuchao. With manual tagging client-go is slightly different. But actually editing client-go's README can be done through staging as well. @caesarxuchao any concerns adding client-go to the list above?

**Comment by caesarxuchao** [comment]
Apart from README and LICENSE, client-go also has .travis.yml, CHANGELOG.md, and INSTALL.md that are not in staging.

Apart from the manual semver tagging, client-go should keep aligned with other repos.

client-go's docs currently needs a few updates. Given that robot is not running daily yet, I want to make one or two more direct merges before move the docs to staging. I'll report here when that's done.

**Comment by sttts** [comment]
> client-go's docs currently needs a few updates. Given that robot is not running daily yet, I want to make one or two more direct merges before move the docs to staging. I'll report here when that's done.

As 1.8.2 is tagged now, I want to start the new push today or tomorrow. @caesarxuchao are you able to do those changes tonight? Otherwise, I would prefer to proceed with the other repos and we align client-go after the push.

**Comment by sttts** [comment]
@eparis https://github.com/kubernetes/kubernetes/pull/54399 is going to be merged soon, @smarterclayton approved. Can you merge the PRs above (https://github.com/kubernetes/kubernetes/issues/54398#issuecomment-338636990) in the external repos?

**Comment by nikhita** [comment]
Ref https://github.com/kubernetes/kubernetes/issues/57559.

**Comment by sttts** [comment]
Can this be closed?

**Comment by nikhita** [comment]
> Can this be closed?

client-go still has authoritative README and LICENSE files.

**Comment by sttts** [comment]
Right. This won't change until the client-go release is integrated officially into the kube release process (with the official authority for the client-go responsibles in the process).

**Comment by nikhita** [comment]
> But actually editing client-go's README can be done through staging as well.

Can we do this now?

We can still update the README through staging every release and then create the release branch manually after the update is done.

> client-go's docs currently needs a few updates. Given that robot is not running daily yet, I want to make one or two more direct merges before move the docs to staging.

@caesarxuchao Do you still want to change the docs?

**Comment by nikhita** [comment]
> We can still update the README through staging every release and then create the release branch manually after the update is done.

From conversation with @sttts : If a release is cut, this would mean that we will need to change the README on the release branch - which adds an extra unwanted dependency.

**Comment by nikhita** [comment]
> This won't change until the client-go release is integrated officially into the kube release process (with the official authority for the client-go responsibles in the process).

Can we close this now? I don't see the client-go release process getting integrated into the kube release process anytime soon. In any case, it deserves a separate issue of it's own and making README and LICENSE from client-go non-authoritative should be a part of that issue.

**Comment by sttts** [comment]
LICSENSE could be moved, couldn't it?

**Comment by nikhita** [comment]
> LICSENSE could be moved, couldn't it?

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/LICENSE It's already moved :)

**Comment by nikhita** [comment]
The only files that are authoritative: CHANGELOG, INSTALL, README?

I think https://github.com/kubernetes/client-go/blob/master/INSTALL.md can also be moved. Wdyt?

**Comment by nikhita** [comment]
/remove-lifecycle stale

created #68790 to move INSTALL.md to staging
